src/pages/PlanPage/hooks/usePlanPageData.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+type URLParams = {
+  owner: string
+  provider: string
+}
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      username: z.string().nullable(),
+      isCurrentUserPartOfOrg: z.boolean(),
+      numberOfUploads: z.number().nullable(),
+    })
+    .nullable(),
+})
+
+export function usePlanPageData() {
+  const { owner, provider } = useParams<URLParams>()
+  const query = `
+      query PlanPageData($username: String!) {
+        owner(username: $username) {
+          username
+          isCurrentUserPartOfOrg
+          numberOfUploads
+        }
+      }
+    `
```

Comment:
`l`: can we move this outside of the hook, that way we it's not being declared upon calling the hook, which is called every time a component re-renders.